---
title: "Marker Development: Allele Frequency Calculations"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## For Primer Pairs L116, L148, L203, L430, and L447
From my first two rounds of marker creation using alignments to the Solidissima Transcriptome.


#### Set Up
Including notes to myself.

```{r library, warning=FALSE, error=FALSE, message=FALSE}
library(tidyverse)
library(knitr)

#Import
samples<-read_csv("allelefreq_spreadsheet.csv")
#kable(head(samples))

#Remove collumns to the right filled with NAs left over from excel
samples <- samples[colSums(!is.na(samples)) > 0]
#Filter rows at bottom with same thing (using tidyverse)
samples <- samples %>% filter_all(any_vars(!is.na(.)))
```

```{r addingalleles}
#going to use mutate
#Similis is 0, Solidissima is 1
#if L116 = hybrid, L116_allele1 = 0, L116_allele2 = 1
#samples %>% mutate(allele = 
#                     ifelse(L116 == "hybrid", 0, 
#                            ifelse(L116 == "similis", 0, 
#                                  ifelse(L116 == "solidissima", 1, NA))))
#cannot easily/effectively turn this into a loop because it is difficult to iterate over collumn names and even more so difficult to create new collumn name within mutate.

#Do over normal for loop
i <- c("L116", "L148","L203","L430","L447")
for (j in i){
  a1 <- paste(j,'allele','1', sep='_')
  samples[[a1]] <- ifelse(samples[[j]] == "hybrid", 0, 
                   ifelse(samples[[j]] == "similis", 0, 
                   ifelse(samples[[j]] == "solidissima", 1, NA)))
  a2 <- paste(j,'allele','2', sep='_')
  samples[[a2]] <- ifelse(samples[[j]] == "hybrid", 1, 
                   ifelse(samples[[j]] == "similis", 0, 
                   ifelse(samples[[j]] == "solidissima", 1, NA)))
}

#Remove full row NAs (from excel document - not all were used for tests)
samples <- samples[which(rowMeans(is.na(samples)) < 0.70), ] #row with 70% or more NA. 15/21 of NA variables (the number of loci*3), is 71%

kable(head(samples))
```

#### Calculations

##### Comparing Summation and Mean Methods

```{r compare}
usingmean <- samples %>%
  group_by(`Mitochondrial Marker`) %>%
  summarize(L116 = (mean(L116_allele_1[!is.na(L116_allele_1)]) + mean(L116_allele_2[!is.na(L116_allele_2)]))/2, 
            L148 = (mean(L148_allele_1[!is.na(L148_allele_1)]) + mean(L148_allele_2[!is.na(L148_allele_2)]))/2, 
            L203 = (mean(L203_allele_1[!is.na(L203_allele_1)]) + mean(L203_allele_2[!is.na(L203_allele_2)]))/2,
            L430 = (mean(L430_allele_1[!is.na(L430_allele_1)]) + mean(L430_allele_2[!is.na(L430_allele_2)]))/2,
            L447 = (mean(L447_allele_1[!is.na(L447_allele_1)]) + mean(L447_allele_2[!is.na(L447_allele_2)]))/2)

#     of L116 allele 1,     in rows that aren't NA,        how many are exactly 1 (solidissima allele)
sum(samples$L116_allele_1[!is.na(samples$L116_allele_1)] == 1)
#     the same for allele 2
sum(samples$L116_allele_2[!is.na(samples$L116_allele_2)] == 1) 
# this is more because hybrids have a 0 for allele 1 and a 1 for allele 2

#the total number of non-na's from the two alleles
sum(!is.na(samples$L116_allele_1)) + sum(!is.na(samples$L116_allele_2))     #119 from each is 238 total

# double check by adding all of the 1 and 0s from both alleles together
sum(samples$L116_allele_1[!is.na(samples$L116_allele_1)] == 1) +
  sum(samples$L116_allele_2[!is.na(samples$L116_allele_2)] == 1) +
  sum(samples$L116_allele_1[!is.na(samples$L116_allele_1)] == 0) +
  sum(samples$L116_allele_2[!is.na(samples$L116_allele_2)] == 0)
# also 238

# Ratio of allele = 1 / allele = not NA
( sum(samples$L116_allele_1[!is.na(samples$L116_allele_1)] == 1) +
  sum(samples$L116_allele_2[!is.na(samples$L116_allele_2)] == 1) ) /
  ( sum(!is.na(samples$L116_allele_1)) + sum(!is.na(samples$L116_allele_2)) )

# Repeat using tidyverse group_by and summarize
usingbasic <- samples %>%
  group_by(`Mitochondrial Marker`) %>%
  summarize(L116 = ( sum(L116_allele_1[!is.na(L116_allele_1)] == 1) +                  # allele 1 = 1
                      sum(L116_allele_2[!is.na(L116_allele_2)] == 1) ) /               # allele 2 = 1
                      ( sum(!is.na(L116_allele_1)) + sum(!is.na(L116_allele_2)) ),     # allele 1 & 2 is 1 or 0
            L148 = ( sum(L148_allele_1[!is.na(L148_allele_1)] == 1) +  
                      sum(L148_allele_2[!is.na(L148_allele_2)] == 1) ) /       
                      ( sum(!is.na(L148_allele_1)) + sum(!is.na(L148_allele_2)) ),
            L203 = ( sum(L203_allele_1[!is.na(L203_allele_1)] == 1) +  
                      sum(L203_allele_2[!is.na(L203_allele_2)] == 1) ) /       
                      ( sum(!is.na(L203_allele_1)) + sum(!is.na(L203_allele_2)) ),
            L430 = ( sum(L430_allele_1[!is.na(L430_allele_1)] == 1) +  
                      sum(L430_allele_2[!is.na(L430_allele_2)] == 1) ) /       
                      ( sum(!is.na(L430_allele_1)) + sum(!is.na(L430_allele_2)) ),
            L447 = ( sum(L447_allele_1[!is.na(L447_allele_1)] == 1) +  
                      sum(L447_allele_2[!is.na(L447_allele_2)] == 1) ) /       
                      ( sum(!is.na(L447_allele_1)) + sum(!is.na(L447_allele_2)) ))
       
kable(usingmean, digits = 4)

kable(usingbasic, digits = 4)
   
```


##### Using Mean to Calculate Frequency

+ Mitochondrial and each locus allele frequencies over all samples.
+ Allele frequencies of each locus splitting into two populations: mitochondrial similis and solidissima.
+ Mitochondrial and locus frequencies at sites and locations.

```{r basiccalc, warning=FALSE, message=FALSE}
#To calculate Solidissima Frequency
# ( Mean(allele1) + Mean(allele2) )
# Because 0,1 0,0 as alleles = 1 solidissima / 4 total = mean = allele freq

#Frequencies of Solidissimas in each populations (ideally 1 in solidissima, and 0 in similis)

byspecies <- samples %>%
  group_by(`Mitochondrial Marker`) %>%
  summarize(L116 = (mean(L116_allele_1[!is.na(L116_allele_1)]) + mean(L116_allele_2[!is.na(L116_allele_2)]))/2, 
            L148 = (mean(L148_allele_1[!is.na(L148_allele_1)]) + mean(L148_allele_2[!is.na(L148_allele_2)]))/2, 
            L203 = (mean(L203_allele_1[!is.na(L203_allele_1)]) + mean(L203_allele_2[!is.na(L203_allele_2)]))/2,
            L430 = (mean(L430_allele_1[!is.na(L430_allele_1)]) + mean(L430_allele_2[!is.na(L430_allele_2)]))/2,
            L447 = (mean(L447_allele_1[!is.na(L447_allele_1)]) + mean(L447_allele_2[!is.na(L447_allele_2)]))/2)

#Formatting
colnames(byspecies)[1] <- ""
byspecies <- byspecies[-1]
rownames(byspecies) <- c("Similis", "Solidissima")
```

**Solidissima Allele Frequency** in each population (as dictated by mitochondrial marker). A 1.0 for Solidissima means that all Solidissima samples according to the mitochondrial marker are homozygous Solidissima, and a 0.0 under Similis means the same for Similis.

``` {r print1, echo=FALSE}
kable(byspecies, digits=3)
```

These values are **significantly** different than what I calculated before. **Please check my math.** These confirm similar relative patterns in effectiveness between the primers (L116 is still bad and L447 is still the best), but all have far closer values for similis detection to expected.

For comparison:
```{r oldcalc, warning=FALSE, message=FALSE, echo=FALSE}
old_freq <- read_csv("oldcalc_freq.csv")
#set headers
rownames(old_freq) <- old_freq$X1
old_freq$X1 <- NULL
#transpose (only using t() results in numeric all becoming strings)
t_freq <- as.data.frame(t(as.matrix(old_freq)))
```

``` {r print2, echo=FALSE}
kable(head(t_freq),digits=3)
```


These were the values that lead us to question if the transcriptome used was important. For clarity, below I have shown difference from ideal. absolute(expected frequency - observed frequency)

```{r substract, warning=FALSE, message=FALSE}
diverge <- byspecies
diverge[2,] <- 1 - diverge[2,]
diverge <- diverge %>%
  rbind(c(diverge$L116[1]-diverge$L116[2],diverge$L148[1]-diverge$L148[2],diverge$L203[1]-diverge$L203[2],diverge$L430[1]-diverge$L430[2],diverge$L447[1]-diverge$L447[2]))
#Similis - Solidissima (positive numbers mean similis is being detected less - higher divergence from expected)
diverge <- full_join(diverge, t_freq)
diverge[5,] <- 1 - diverge[5,]
diverge <- diverge %>%
  rbind(c(diverge$L116[4]-diverge$L116[5],diverge$L148[4]-diverge$L148[5],diverge$L203[4]-diverge$L203[5],diverge$L430[4]-diverge$L430[5],diverge$L447[4]-diverge$L447[5]))

rownames(diverge) <- c("New Sim (O-E)", " New Sol (O-E)", "Difference in Detection", "Old Sim (O-E)", "Old Sol (O-E)", "Old Difference in Detection")

kable(diverge, digits = 2)

```

Note here than in rows 3 and 6, *postive* values for difference in detection mean that similis is being detected less well at this locus than solidissima. In the newly calculated values, there is a fairly even split between positive and negative values (to me indicating that there is not a bias against similis), whereas in the older differences in detection values all are postive and very large (indicating the anti-similis bias that we previously concluded).

I have looked through my excel calculations from earlier and have still not found the error yet but clearly this was a better way to do the calculation if you agree with my math.

### By population
```{r bysite}
bysite <- samples %>%
  group_by(`Site`) %>%
  summarize(L116 = (mean(L116_allele_1[!is.na(L116_allele_1)]) + mean(L116_allele_2[!is.na(L116_allele_2)]))/2, 
            L148 = (mean(L148_allele_1[!is.na(L148_allele_1)]) + mean(L148_allele_2[!is.na(L148_allele_2)]))/2, 
            L203 = (mean(L203_allele_1[!is.na(L203_allele_1)]) + mean(L203_allele_2[!is.na(L203_allele_2)]))/2,
            L430 = (mean(L430_allele_1[!is.na(L430_allele_1)]) + mean(L430_allele_2[!is.na(L430_allele_2)]))/2,
            L447 = (mean(L447_allele_1[!is.na(L447_allele_1)]) + mean(L447_allele_2[!is.na(L447_allele_2)]))/2)
#Set NAs to be a - to reduce clutter
options(knitr.kable.NA = '-')
kable(bysite, digits=1)
```

Showing Solidissima allele frequency (a 1.0 means that all those were solidissima, a 0.0 means all similis)

These are condensed into broader areas below.

```{r byloc, echo=FALSE}
byloc <- samples %>%
  group_by(`Location`) %>%
  summarize(L116 = (mean(L116_allele_1[!is.na(L116_allele_1)]) + mean(L116_allele_2[!is.na(L116_allele_2)]))/2, 
            L148 = (mean(L148_allele_1[!is.na(L148_allele_1)]) + mean(L148_allele_2[!is.na(L148_allele_2)]))/2, 
            L203 = (mean(L203_allele_1[!is.na(L203_allele_1)]) + mean(L203_allele_2[!is.na(L203_allele_2)]))/2,
            L430 = (mean(L430_allele_1[!is.na(L430_allele_1)]) + mean(L430_allele_2[!is.na(L430_allele_2)]))/2,
            L447 = (mean(L447_allele_1[!is.na(L447_allele_1)]) + mean(L447_allele_2[!is.na(L447_allele_2)]))/2)

kable(byloc, digits=1)
```

+ Where George's Bank is just GBE - expected solidissima (1.0).
+ Georgia is just GA - expected similis (0.0).
+ Massachusetts, PT and MW - mixed.
+ Northern Long Islang - mostly similis (0.0).
+ Southern Long Island - mostly solidissima (1.0).
+ Reisman Hatery - mixed, more solidissima.

### For (practice) extractions - Which samples might I need more of anyway because they showed interesting patterns
```{r which}
#stats and dplyr both have filter functions and I have both loaded?

hyb <- filter(samples, L203 == "hybrid" | L430 == "hybrid" | L447 == "hybrid")

```


### Hybrid Index
```{r index}

samples %>% 
  mutate(hybrid_index = mean(L116_allele_1,L116_allele_2,
                             L148_allele_1,L148_allele_2,
                             L203_allele_1,L203_allele_2,
                             L430_allele_1,L430_allele_2,
                             L447_allele_1,L447_allele_2, na.rm=TRUE))

samples_hi <- mutate(samples, hybrid_index = rowMeans(select(samples, contains("allele")), na.rm = TRUE))
samples_hi_sub <- samples_hi[-grep("Reitsman", samples_hi$Location), ]


samples_hi_sub[which(samples_hi_sub[,3]=="Northern Long Island"),3] <- "Long Island North Shore"
samples_hi_sub[which(samples_hi_sub[,3]=="Southern Long Island"),3] <- "Long Island South Shore"

#nrow(samples_hi_sub[str_detect(samples_hi_sub1$Location, "North"), ])

set.seed(6619);samples_hi_sub %>%
   mutate(Location = factor(Location, levels=c("Georgia", "Long Island South Shore", "Long Island North Shore", "Massachusetts",  "George's Bank"))) %>%
  ggplot(aes(hybrid_index, Location, colour = `Mitochondrial Marker`)) + 
  geom_jitter(width = 0.05, height = 0.2) +
  scale_color_manual(values=c("#ff9715ff", "#8be067ff"))

samples_hi_sub1 <- samples_hi_sub
samples_hi_sub1$`Mitochondrial Marker`[84] <- 'microsatellite admixed'
samples_hi_sub1$`Mitochondrial Marker`[92] <- 'microsatellite admixed'
samples_hi_sub1$`Mitochondrial Marker`[95] <- 'microsatellite admixed'
samples_hi_sub1$`Mitochondrial Marker`[109] <- 'microsatellite admixed'

set.seed(6619);samples_hi_sub1 %>%
   mutate(Location = factor(Location, levels=c("Georgia", "Long Island South Shore", "Long Island North Shore", "Massachusetts",  "George's Bank"))) %>%
  ggplot(aes(hybrid_index, Location, colour = `Mitochondrial Marker`)) + 
  geom_jitter(width = 0.05, height = 0.2) +
  scale_color_manual(values=c("#f20253ff", "#ff9715ff", "#a4fd7eff"))

#Put in order: 6619, 234
#GBE, Mass, n LI, s LI, GA
   

samples_hi_sub1[str_detect(samples_hi_sub1$Location, "North"), ]
```
```{r exportreimport}

#For speed
#write.csv(samples,"hybridallele.csv")
hybridsamples<-read_csv("hybridallele2.csv")
hybridsamples[which(hybridsamples[,3]=="Northern Long Island"),3] <- "Long Island North Shore"
hybridsamples[which(hybridsamples[,3]=="Southern Long Island"),3] <- "Long Island South Shore"
hybridsamples <- hybridsamples[-grep("Reitsman", hybridsamples$Location), ]


set.seed(699);hybridsamples %>%
   mutate(Location = factor(Location, levels=c("Georgia", "Long Island South Shore", "Long Island North Shore", "Massachusetts",  "George's Bank"))) %>%
  ggplot(aes(`Hybrid Index`, Location, colour = `Mitochondrial Marker`)) + 
  geom_jitter(width = 0.02, height = 0.2,size=3) +
  scale_color_manual(values=c("#ff9715ff", "#8be067ff")) +
  theme_minimal(base_size = 22) +
  xlim(0,1)

samples_hi_sub1 <- hybridsamples
samples_hi_sub1$`Mitochondrial Marker`[84] <- 'microsatellite admixed'
samples_hi_sub1$`Mitochondrial Marker`[85] <- 'microsatellite admixed'
samples_hi_sub1$`Mitochondrial Marker`[92] <- 'microsatellite admixed'
samples_hi_sub1$`Mitochondrial Marker`[95] <- 'microsatellite admixed'
samples_hi_sub1$`Mitochondrial Marker`[109] <- 'microsatellite admixed'

set.seed(699);samples_hi_sub1 %>%
   mutate(Location = factor(Location, levels=c("Georgia", "Long Island South Shore", "Long Island North Shore", "Massachusetts",  "George's Bank"))) %>%
  ggplot(aes(`Hybrid Index`, Location, colour = `Mitochondrial Marker`)) + 
  geom_jitter(width = 0.02, height = 0.2, size=3) +
  scale_color_manual(values=c("#f20253ff", "#ff9715ff", "#a4fd7eff")) +
  theme_minimal(base_size = 22) +
  xlim(0,1)
```



```{r number of data points}
samples_hi_sub[str_detect(samples_hi_sub1$Location, "North"), ] %>%
  ggplot(aes(hybrid_index, Location, colour = `Mitochondrial Marker`)) + 
  geom_jitter(width = 0.01, height = 0.2) +
  scale_color_manual(values=c("#ff9715ff", "#8be067ff"))
```