---
title: "Pseudo Code Plan - Spisula Primers"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#### Done

* Assemble Transcriptome
* Align Sequences
  - Group all Sim and all Sol together
* Variant Calling

#### To Be Done

  * (Filter Transcriptome before Variant Calling to have one read per gene)
* Filter 
  - Sim by sites that are non-heterozygous by different from reference 1/1
  - Compare to Sol sites to find Sol that is non-heterozygous and matches reference at those postions
* Cross-reference those positions with exon edges.
* Cross-reference ones within exons with known cheap primers.


